src/Controller/UserController.php

```diff
@@ -59,63 +62,82 @@ public function viewUser(User $user = null, UserRepository $userRepository, Seri
+                            ClientRepository $repository)
     {
-        $data = json_decode($request->getContent(), true);
-
-        $clientId = $data['client']['id'];
         $client = $repository->findClient($clientId);
+        $user->setClient($client[0]);
 
-        if ($client) {
-            $user = new User();
-            $user->createUserObject($data);
-            $user->setClient($client[0]);
+        $validatorResults = $validator->validate($user, null, null);
+        if(count($validatorResults) > 0){
+            return $this->view($validatorResults, Response::HTTP_BAD_REQUEST);
+        }
 
-            $manager->persist($user);
-            $manager->flush();
+        $manager->persist($user);
+        $manager->flush();
 
-            return $user;
-        }
+        return $this->view($user, Response::HTTP_CREATED);
     }
 
     /**
      * @Rest\Put(
-     *     path = "/modifyuser/{userId}",
+     *     path = "/users/{userId}",
      *     name = "modify_user")
-     * @param Request $request
-     * @param EntityManagerInterface $manager
-     * @param ClientRepository $repository
-     * @param UserRepository $userRepository
-     * @return User
+     * @ParamConverter("user", converter="fos_rest.request_body")
+     * @param $userId
+     * @param UserRepository $repository
+     * @return View
      */
-    public function modifyUser(Request $request, EntityManagerInterface $manager, UserRepository $userRepository, $userId)
+    public function modifyUser(User $user, $userId, UserRepository $repository, EntityManagerInterface $manager)
     {
-        $data = json_decode($request->getContent(), true);
+        $registeredUser = $repository->findUser($userId);
 
-        $user = ($userRepository->findUser($userId))[0];
+        if(empty($registeredUser)){
+            return new View('Cet utilisateur n\'existe pas', Response::HTTP_NOT_FOUND);
+        }
+
+        $registeredUser=$registeredUser[0];
+        $registeredUser
+            ->setUsername($user->getUsername())
+            ->setEmail($user->getEmail())
+            ->setPassword($user->getPassword());
 
-        if ($user){
-            if($data['username']){
-                $user->setUsername($data['username']);
-            }
-            if ($data['password']){
-                $user->setPassword($data['password']);
-            }
-            if ($data['email']){
-                $user->setEmail($data['email']);
-            }
+        $manager->persist($registeredUser);
+        $manager->flush();
 
-            $manager->persist($user);
-            $manager->flush();
+        return $this->view($registeredUser, Response::HTTP_CREATED);
+    }
+
+    /**
+     * @Rest\Delete(
+     *     path = "/users/{userId}",
+     *     name = "delete_user")
+     * @param $userId
+     * @param UserRepository $repository
+     * @return View
+     */
+    public function deleteUser($userId, UserRepository $repository, EntityManagerInterface $manager)
```

Comment:
Même principe ici, la route doit être /users/{id}

Reply:
It's already the case. The id's called userId to distinguish it from ClientId.